vsintegration/src/FSharp.Editor/CodeFixes/ChangeEqualsInFieldTypeToColon.fs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System.Collections.Immutable
+
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Text
+open Microsoft.CodeAnalysis.CodeFixes
+
+open FSharp.Compiler.Text
+
+open CancellableTasks
+
+[<ExportCodeFixProvider(FSharpConstants.FSharpLanguageName, Name = CodeFix.ChangeEqualsInFieldTypeToColon)>]
+type internal ChangeEqualsInFieldTypeToColonCodeFixProvider() =
+    inherit CodeFixProvider()
+
+    let isInRecord (document: Document) (range: range) =
+        cancellableTask {
+            let! parseResults = document.GetFSharpParseResultsAsync(nameof ChangeEqualsInFieldTypeToColonCodeFixProvider)
+
+            return parseResults.IsPositionWithinRecordDefinition(range.Start)
+        }
+
+    override _.FixableDiagnosticIds = ImmutableArray.Create("FS0010")
+
+    override this.RegisterCodeFixesAsync context =
+        // This is a performance shortcut.
+        // Since FS0010 fires all too often, we're just stopping any handling of it
+        // in case when there is the message doesn't have a notion of symbols in question.
+        // The code fix logic itself still has this logic and implements it more reliably.
+        if
+            context.Diagnostics
```

Comment:
These are still (IMO) unnecessary allocations for the seq enumerator and for string building.
Why not a direct exact string equality check for the full string? We know how FS0010 looks like (and can even cache the localized version of the .Diagnostics fields contains the localized one, not sure know).

Reply:
You do not have to do Seq-based enumeration which needs a heap-allocated Enumerator.
The same for the internal allocations needed to do string interpolation, even though we know the strings.

The FS0010 strings to compare can be directly referenced within the  codebase and not repeated, to eliminate the fear of those texts changing. (and the right characters passed in as a parameters, and then the whole string statically cached for example)

Reply:
@T-Gro so actually it's just occurred to me that we don't have to iterate through anything here. The way the code fixes work, this can will be triggered only for one fs0010 diagnostic. As in, the code fix context doesn't contain diagnostics that the code fix is not made for:
![image](https://github.com/dotnet/fsharp/assets/5451366/febda447-8efd-470e-b44c-643037fb46f5)

So we can access the message right away, counting with the fact it's very unlikely anybody adds another diagnostic here.